ckanext/s3filestore/commands.py

```diff
@@ -52,5 +67,62 @@ def check_config(self):
             print(str(ex))
             sys.exit(1)
 
-
         print('Configuration OK!')
+
+    def upload_all(self):
+        BASE_PATH = config.get('ckan.storage_path', '/var/lib/ckan/default/resources')
+        SQLALCHEMY_URL = config.get('sqlalchemy.url', 'postgresql://user:pass@localhost/db')
+        if config.get('ckanext.s3filestore.aws_use_ami_role', False):
+            AWS_ACCESS_KEY_ID = None
+            AWS_SECRET_ACCESS_KEY = None
+        else:
+            AWS_ACCESS_KEY_ID = config.get('ckanext.s3filestore.aws_access_key_id')
+            AWS_SECRET_ACCESS_KEY = config.get('ckanext.s3filestore.aws_secret_access_key')
+        AWS_BUCKET_NAME = config.get('ckanext.s3filestore.aws_bucket_name')
+        AWS_S3_ACL = config.get('ckanext.s3filestore.acl', 'public-read')
+        resource_ids_and_paths = {}
+
+        for root, dirs, files in os.walk(BASE_PATH):
+            if files:
+                resource_id = root.split('/')[-2] + root.split('/')[-1] + files[0]
+                resource_ids_and_paths[resource_id] = os.path.join(root, files[0])
+
+        print('Found {0} resource files in the file system'.format(
+            len(resource_ids_and_paths.keys())))
+
+        engine = create_engine(SQLALCHEMY_URL)
+        connection = engine.connect()
+
+        resource_ids_and_names = {}
+
+        try:
+            for resource_id, file_path in resource_ids_and_paths.iteritems():
+                resource = connection.execute(text('''
+                    SELECT id, url, url_type
+                    FROM resource
+                    WHERE id = :id
+                '''), id=resource_id)
+                if resource.rowcount:
+                    _id, url, _type = resource.first()
+                    if _type == 'upload' and url:
+                        file_name = url.split('/')[-1] if '/' in url else url
+                        resource_ids_and_names[_id] = file_name.lower()
+        finally:
+            connection.close()
+            engine.dispose()
+
+        print('{0} resources matched on the database'.format(
+            len(resource_ids_and_names.keys())))
+
+        s3_connection = boto3.resource('s3', aws_access_key_id=AWS_ACCESS_KEY_ID, aws_secret_access_key=AWS_SECRET_ACCESS_KEY)
+
+        uploaded_resources = []
+        for resource_id, file_name in resource_ids_and_names.iteritems():
+            key = 'resources/{resource_id}/{file_name}'.format(
+                resource_id=resource_id, file_name=file_name)
```

Comment:
@ThrawnCA , With this change, are we sure it won't override files which already exist on s3?

Reply:
It will overwrite them, yes. Probably possible to avoid that.

Reply:
@ThrawnCA  well thats not very good, the pairtree migrate at least checked before it copied and skipped if it already existed. With the way s3 resources are stored, someone could update the file with the same filename so it would be overridden with the old version of said file which is no good. 

We will need to fix this before we run this in real life just in case, are you ok with that?

Reply:
Yep, can implement something today.

Reply:
Updated in #41 to try a HEAD on the object before uploading.